be/src/storage/lake/lake_persistent_index.cpp

```diff
@@ -300,12 +302,14 @@ Status LakePersistentIndex::major_compact(int64_t min_retain_version, TxnLogPB*
     RETURN_IF_ERROR(merge_sstables(std::move(iter_ptr), &builder));
     RETURN_IF_ERROR(wf->close());
 
-    auto max_compaction_versions = config::lake_pk_index_sst_max_compaction_versions;
+    auto max_compaction_bytes = config::lake_pk_index_sst_max_compaction_bytes;
+    size_t total_filesize = 0;
     for (const auto& sstable : _sstables) {
         auto input_sstable = txn_log->mutable_op_compaction()->add_input_sstables();
         auto sstable_pb = sstable->sstable_pb();
         input_sstable->CopyFrom(sstable_pb);
-        if (txn_log->op_compaction().input_sstables_size() >= max_compaction_versions) {
+        total_filesize += sstable_pb.filesize();
+        if (total_filesize >= max_compaction_bytes) {
```

Comment:
what if the first sstable size exceeds the max_compaction_bytes? only one sstable will be chosen, and what will happen for the remaining workflow, and what is expected to do if this happens in production, adjusting the parameter? to what number?

Reply:
It should also meet the `lake_pk_index_sst_min_compaction_versions` threshold.